build/dsc/templates/dsc_type.rb.erb

```diff
@@ -74,28 +74,24 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   <%  end -%>
   end
 
-  newproperty(:dscmeta_resource_friendly_name) do
+  newparam(:dscmeta_resource_friendly_name) do
     desc "A read-only value that is the DSC Resource Friendly Name ('<%= resource.friendlyname %>')."
 
-    def retrieve
-      '<%= resource.friendlyname %>'
-    end
-
     validate do |value|
-      fail 'dscmeta_resource_friendly_name is read-only'
+      fail 'dscmeta_resource_friendly_name is read-only' if value != '<%= resource.friendlyname %>'
```

Comment:
OK - so I think we can actually simplify this whole thing quite a bit, but departing from prior art.  Let me PR your PR to show you what I mean.